pkg/eks/auth.go

```diff
@@ -246,12 +205,13 @@ func (c *ClientConfig) NewClientSet() (*clientset.Clientset, error) {
 	return client, nil
 }
 
-// NewClientSetWithEmbeddedToken creates a new API client with an embedded STS token
-func (c *ClientConfig) NewClientSetWithEmbeddedToken() (*clientset.Clientset, error) {
-	clientConfig, err := c.WithEmbeddedToken()
+// NewClientSet creates a new API client in one go with an embedded STS token
+func (c *ClusterProvider) NewClientSet(spec *api.ClusterConfig) (*clientset.Clientset, error) {
```

Comment:
Code would be easier to read and understand if this was NewClientSetWithStsToken (or similar).  There is a NewClientSet() on ClientConfig and ClusterProvider

Reply:
True, I just went for shorter name.

Reply:
I'll make it `NewStdClientSet`.